modules/tasks/tasks/src/main/resources/webapp/js/controllers.js

```diff
@@ -206,61 +207,67 @@
+                activities = Activities.query(function () {
+                    var item, i, j;
 
-                        for (i = 0; i < tasks.length; i += 1) {
-                            item = {
-                                task: tasks[i],
-                                success: $scope.getNumberOfActivities(tasks[i].id, 'SUCCESS'),
-                                error: $scope.getNumberOfActivities(tasks[i].id, 'ERROR')
-                            };
-                        }
+                    for (i = 0; i < tasks.length; i += 1) {
+                        item = {
+                            task: tasks[i],
+                            success: $scope.getNumberOfActivities(tasks[i].id, 'SUCCESS'),
+                            error: $scope.getNumberOfActivities(tasks[i].id, 'ERROR')
+                        };
+                    }
 
-                        for (i = 0; i < RECENT_TASK_COUNT && i < activities.length; i += 1) {
-                            for (j = 0; j < tasks.length; j += 1) {
-                                if (activities[i].task === tasks[j].id) {
-                                    $scope.activities.push({
-                                        task: activities[i].task,
-                                        trigger: tasks[j].trigger,
-                                        actions: tasks[j].actions,
-                                        date: activities[i].date,
-                                        type: activities[i].activityType,
-                                        name: tasks[j].name
-                                    });
-                                    break;
-                                }
+                    for (i = 0; i < RECENT_TASK_COUNT && i < activities.length; i += 1) {
+                        for (j = 0; j < tasks.length; j += 1) {
+                            if (activities[i].task === tasks[j].id) {
+                                $scope.activities.push({
+                                    task: activities[i].task,
+                                    trigger: tasks[j].trigger,
+                                    actions: tasks[j].actions,
+                                    date: activities[i].date,
+                                    type: activities[i].activityType,
+                                    name: tasks[j].name
+                                });
+                                break;
                             }
                         }
-                    });
+                    }
                 });
-            };
-            $scope.getTasks();
+            });
+        };
+        $scope.getTasks();
 
-        });
+        $scope.logsDisabled = function() {
+            return ($rootScope.settings.taskLogActivities === "None");
+        };
+        $scope.successLogsDisabled = function() {
+            return ($rootScope.settings.taskLogActivities === "Only failure");
+        };
```

Comment:
Could these methods (logsDisabled and successLogsDisabled) be moved to rootscope? I see they are declared in two controllers - this is redundant.